client/auth.py

```diff
@@ -0,0 +1,296 @@
+            auth_user_change_password_request=AuthUserChangePasswordRequest(name=name, hashedPassword=hashed_password),
+        )
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_change_password_response
+
+    async def user_grant_role(self, user: str, role: str) -> AuthUserGrantRoleResponse:
+        """
+        Grant role for a user.
+        """
+        request_with_token = RequestWithToken(
+            token=self.token, auth_user_grant_role_request=AuthUserGrantRoleRequest(user=user, role=role)
+        )
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_grant_role_response
+
+    async def user_revoke_role(self, name: str, role: str) -> AuthUserRevokeRoleResponse:
+        """
+        Revoke role for a user.
+        """
+        request_with_token = RequestWithToken(
+            token=self.token, auth_user_revoke_role_request=AuthUserRevokeRoleRequest(name=name, role=role)
+        )
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_user_revoke_role_response
+
+    async def role_add(self, name: str) -> AuthRoleAddResponse:
+        """
+        Adds role.
+        """
+        if name == "":
+            msg = "role name is empty"
+            raise Exception(msg)
+
+        request_with_token = RequestWithToken(token=self.token, auth_role_add_request=AuthRoleAddRequest(name=name))
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_role_add_response
+
+    async def role_get(self, role: str) -> AuthRoleGetResponse:
+        """
+        Gets role.
+        """
+        request_with_token = RequestWithToken(token=self.token, auth_role_get_request=AuthRoleGetRequest(role=role))
+        er = await self.handle_req(req=request_with_token, use_fast_path=True)
+        return er.auth_role_get_response
+
+    async def role_list(self) -> AuthRoleListResponse:
+        """
+        Lists role.
+        """
+        request_with_token = RequestWithToken(token=self.token, auth_role_list_request=AuthRoleListRequest())
+        er = await self.handle_req(req=request_with_token, use_fast_path=True)
+        return er.auth_role_list_response
+
+    async def role_delete(self, role: str) -> AuthRoleDeleteResponse:
+        """
+        Deletes role.
+        """
+        request_with_token = RequestWithToken(
+            token=self.token, auth_role_delete_request=AuthRoleDeleteRequest(role=role)
+        )
+        er = await self.handle_req(req=request_with_token, use_fast_path=False)
+        return er.auth_role_delete_response
+
+    async def role_grant_permission(
+        self, name: str, perm_type: PermissionType, key: bytes, range_end: bytes | None = None
+    ) -> AuthRoleGrantPermissionResponse:
+        """
+        Grants role permission.
+        """
+        if perm_type is None:
```

Comment:
The permission type cannot be None.

Reply:
You are right. param `perm_type` and `key` isn't option.